xarray/backends/netCDF4_.py

```diff
@@ -348,6 +348,7 @@ def open(
         lock=None,
         lock_maker=None,
         autoclose=False,
+        memory=None,
```

Comment:
```suggestion
        *,
        memory=None,
```
I wonder if we should make this argument (and eventually the rest) keyword only?

Reply:
Marking all arguments as keyword-only would be a braking change?

Reply:
Could use our [new decorator](https://github.com/pydata/xarray/blob/3d3b236df3f8f5982760e68e064eb0db2aacc4a2/xarray/util/deprecation_helpers.py#L44), however, I would suggest to do this outside of this PR? (Also is this actually public API? - isn't this accessed via `xr.open_dataset(..., backend_kwargs=dict(...))`?

Reply:
@mathause I also think this could be done in another PR.

Reply:
Yes, making all arguments keyword only now is a breaking change and outside the scope of this PR. But I think making `engine` keyword-only should be done now so users won't have to change their code when we do the switch.